timesketch/api/v1/resources.py

```diff
@@ -695,10 +700,140 @@ def post(self, sketch_id):
 class AggregationResource(ResourceMixin, Resource):
     """Resource to query for aggregated results."""
 
+    @login_required
+    def get(self, sketch_id, aggregation_id):  # pylint: disable=unused-argument
+        """Handles POST request to the resource.
+
+        Handler for /api/v1/sketches/:sketch_id/aggregation/:aggregation_id
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+            aggregation_id: Integer primary key for an agregation database model
+
+        Returns:
+            JSON with aggregation results
+        """
+        # TODO: Implement once aggregations are saved in the datastore.
+        return {}
+
+    @login_required
+    def post(self, sketch_id, aggregation_id):  # pylint: disable=unused-argument
+        """Handles POST request to the resource.
+
+        Handler for /api/v1/sketches/:sketch_id/aggregation/:aggregation_id
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+            aggregation_id: Integer primary key for an agregation database model
+        """
+        # TODO: Implement once we have an aggregation model in the datastore.
+        abort(HTTP_STATUS_CODE_FORBIDDEN)
+
+
+class AggregationExploreResource(ResourceMixin, Resource):
+    """Resource to send an aggregation request."""
+
+    RESULT_FIELDS = frozenset(['_shards', 'hits', 'timed_out', 'took'])
+
+    @login_required
+    def post(self, sketch_id):
+        """Handles POST request to the resource.
+
+        Handler for /api/v1/sketches/<int:sketch_id>/aggregation/explore
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+
+        Returns:
+            JSON with aggregation results
+        """
+        form = AggregationExploreForm.build(request)
+        if not form.validate_on_submit():
+            return abort(HTTP_STATUS_CODE_BAD_REQUEST)
+
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        sketch_indices = {
+            t.searchindex.index_name
+            for t in sketch.timelines
+        }
+
+        aggregation_dsl = form.aggregation_dsl.data
+        aggregator_name = form.aggregator_name.data
+        aggregator_parameters = form.aggregator_parameters.data
+
+        if aggregator_name and aggregator_parameters:
+            agg_class = aggregator_manager.AggregatorManager.get_aggregator(
+                aggregator_name)
+            if not agg_class:
+                return {}
+            if not aggregator_parameters:
+                aggregator_parameters = {}
+            aggregator = agg_class(sketch_id=sketch_id)
+            time_before = time.time()
```

Comment:
You get this in the response back in the "took" value. Why calculating this yourself?

Reply:
because this is running an aggregator class, and they return back an AggregationResult object, which does not have that stored... and aggregators may just run that aggregation logic in code and not rely on ES at all, so there is no guarantee that a counter is maintained there

Reply:
ok, got it.